### PR TITLE
Agg functions supports enum Expr (implements issue #423)

### PR DESCRIPTION
**Pull request by ghost**
This PR implements the ability for aggregate expr expressions such as :
SUM(id + 1)
MAX(col + col2)
SUM(id * rate) 

implements improvements stated in issue #423

### CODE REVIEW

**Comment by panarch** [comment]
Awesome!
I saw that you consider how to deal with `Value::Null` in `SUM` aggregation.
That is already defined in the existing tests.

https://github.com/gluesql/gluesql/blob/69c9e8445255e3a24fddb3820a948337122a645f/test-suite/src/aggregate.rs#L48-L53

Someone who wants to ignore nulls then they can use `WHERE age IS NOT NULL`
```sql
SELECT SUM(age), MAX(age), MIN(age) FROM Item WHERE age IS NOT NULL
```

Only except for `COUNT(*)`, there's no exceptional handlers for aggregate functions.

**Comment by earney** [comment]
I will continue this PR once the Show Indexes PR is merged.

File: core/src/executor/aggregate/state.rs [hunk 1]
```diff
@@ -175,39 +177,87 @@ impl<'a> State<'a> {
...
+                    BinaryOperator::Minus => left_value.subtract(right_value),
+                    BinaryOperator::Multiply => left_value.multiply(right_value),
+                    BinaryOperator::Divide => left_value.divide(right_value),
+                    BinaryOperator::Modulo => left_value.modulo(right_value),
+                    _ => Err(AggregateError::UnsupportedAggregateFunction.into()),
+                }
+            }
+            _ => Err(AggregateError::UnsupportedAggregateFunction.into()),
+        }
+    }
```
Comment (panarch) [inline core/src/executor/aggregate/state.rs:245]: Aggregate evaluation is not supposed to be done in here.
You may need to use
https://github.com/gluesql/gluesql/blob/e4c04c1135680b6199b965d732591029f46d808e/core/src/executor/evaluate/mod.rs#L27-L33

Comment (panarch) [reply core/src/executor/aggregate/state.rs:245]: > so we will need to supply a storage variable,
yes I think so. Corresponding interface change would be required for `aggregate/state.rs`

**Comment by earney** [comment]
Making the requested changes, I believe are a bit beyond by current skill level, so I'm gonna put this on pause.  If someone is interested in implementing this, please feel free.

**Pull request closed without merging**